lib/isolator/isolate.rb

```diff
@@ -4,9 +4,9 @@ module Isolator
   # Add .isolate function to build and register adapters
   module Isolate
     def isolate(id, **options)
-      raise "Adapter already registered: #{id}" if Isolator.adapters.key?(id.to_s)
-      adapter = AdapterBuilder.call(**options)
-      Isolator.adapters[id.to_s] = adapter
+      AdapterBuilder.call(**options).tap do |adapter|
```

Comment:
We should raise error before `.call`, 'cause we're patching a target module in `.call`.